### ISSUE TITLE
MeshLine.load submeshes & boundaries

### ISSUE THREAD

**Issue by gdmcbain** [kinnala/scikit-fem#175]
`MeshLine.load` warns

> UserWarning: Unable to load submeshes

for any input, and fails to read any subdomains and boundaries present.

The immediate cause is that `'line'` is absent from the keys in

https://github.com/kinnala/scikit-fem/blob/dadd4bac13ca66786c6a545fbb3f139c8b34836a/skfem/mesh/mesh.py#L286-L291

While the one-dimensional examples to date just use `MeshLine.__init__`, and this suffices for most purposes, it is occasionally useful to generate even one-dimensional meshes externally in Gmsh; e.g. for problems like the ‘composite wall composed of _n_ slabs of thicknesses _l1, …, ln_ and conductivities _K1, …, Kn_’ of Carslaw & Jaeger (1959, _Conduction of Heat in Solids_, §3.2 ‘Steady temperature’, Oxford University Press).

Minimal example: [line.geo.gz](https://github.com/kinnala/scikit-fem/files/3155517/line.geo.gz).

```
Point(1) = {0, 0, 0};
Point(2) = {1, 0, 0};
Line(1) = {1, 2};
```

Generate mesh with
```shell
gmsh -1 line.geo
```
and attempt to load with
```python
MeshLine.load('line.msh')
```

**Comment by gdmcbain** [comment]
Just adding `'line': 'vertex'` seems to be enough to get `MeshLine.load` to not warn and to successfully read the subdomain defined by the addition line in `line.geo`
```
Physical Curve("domain") = {1};
```

**Comment by gdmcbain** [comment]
`MeshLine.load` is not loading the boundaries correctly; e.g. for

```
Point(1) = {0, 0, 0};
Point(2) = {1, 0, 0};
Line(1) = {1, 2};
Physical Point("left") = {1};
Physical Point("right") = {2};
Physical Curve("domain") = {1};
```

which produces [line.msh.gz](https://github.com/kinnala/scikit-fem/files/3160026/line.msh.gz), containing
```
$PhysicalNames
3
0 1 "left"
0 2 "right"
1 3 "domain"
$EndPhysicalNames
```

**Comment by gdmcbain** [comment]
I'm wondering whether the problem follows from `MeshLine._build_mappings` sorting `.p` whereas that method of other descendents, e.g. `MeshTri`, `MeshTet`, doesn't.

**Comment by gdmcbain** [comment]
Yes, this is the problem.  In the above example, at

https://github.com/kinnala/scikit-fem/blob/9ff1fd5c76a5b3044df563721c0c80fe6069e651/skfem/mesh/mesh.py#L314-L316

we have `facets == array([[0], [1]])`, but `bndfacets == array([0, 5])`, i.e. `facets` is numbered as in `line.msh` but `bndfacets` uses the reordered nodal indices.

Then `dic == {{(0,): 1, (1,): 2}` (input order) so in 

https://github.com/kinnala/scikit-fem/blob/9ff1fd5c76a5b3044df563721c0c80fe6069e651/skfem/mesh/mesh.py#L324-L326

the `if`-clause does get `'left'` but only by luck since node 0 survived the permutation but misses `'right'` since `self.external.points[1]` at _x_ = 1.0 has been renumbered as `self.p[:, 5]`; i.e. `index` only contains one entry
```python
array([[1, 0]])
```

I don't know how to fix this.  I don't think it can be done at this point of `_parse_submeshes` since the permutation of the points is not remembered.

**Comment by kinnala** [comment]
I can't recall now the exact reason for doing the sort in `_build_mappings`. There could have been some technical problem but maybe its only because `mesh.p` was easier to interpret this way.

**Comment by kinnala** [comment]
I think you could as a workaround (for 1D meshes) re-sort `self.external.points` in `_parse_submeshes` and by that way obtain the correspondence 0<->0, 1<->5 by setting the kwarg `return_index` or `return_inverse` to `True` in `np.sort`.

**Comment by gdmcbain** [comment]
> I can't recall now the exact reason for doing the sort in _build_mappings. There could have been some technical problem but maybe its only because mesh.p was easier to interpret this way.

That sorting is relied upon in a couple of places: 
* the argument `t` that is passed to `MeshLine.__init__` is always silently ignored and
* `MeshLine.t` and `MeshLine.facets` are defined in terms of the sorted `MeshLine.p`.

https://github.com/kinnala/scikit-fem/blob/625ac42ed2185982752c49def5259c6f4e6fa4a2/skfem/mesh/mesh_line.py#L44-L47

**Comment by gdmcbain** [comment]
So if it's O. K. to drop the sorting of the `p`, there's a fix in #176.

**Comment by kinnala** [comment]
Nothing left here I presume.
